infra/modules/aws-cognito-pool/main.tf

```diff
@@ -0,0 +1,45 @@
+resource "aws_cognito_identity_pool" "main" {
```

Comment:
Creating a cognito identity pool; we will use it to create an identity and then use that identity to request a token from lambda side.